Add pt-br translation (#6310)
mdn/translated-content: files/pt-br/mozilla/add-ons/webextensions/api/browsingdata/removecache/index.md

---
title: browsingData.removeCache()
slug: Mozilla/Add-ons/WebExtensions/API/browsingData/removeCache
tags:
  - API
  - Add-ons
  - Extensions
  - Method
  - Reference
  - WebExtensions
  - browsingData
  - removeCache
browser-compat: webextensions.api.browsingData.removeCache
translation-of: Mozilla/Add-ons/WebExtensions/API/browsingData/removeCache
---
{{AddonSidebar()}}

Limpa o cache do navegador.

> Note que apesar desta função receber um objeto {{WebExtAPIRef("browsingData.RemovalOptions")}}, isso será ignorado. Todo o cache sempre será limpo usando esta função.

Reviewer: ```suggestion
Note que apesar desta função receber um objeto {{WebExtAPIRef("browsingData.RemovalOptions")}}, ele será ignorado. Todo o cache sempre será limpo usando esta função.
```

Reviewer: Mudando pra leitura ficar mais natural, o que você acha?

> Esta é uma função assíncrona que retorna uma [`Promessa`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise).

Reviewer: ```suggestion
Esta é uma função assíncrona que retorna uma [`Promise`](/pt-BR/docs/Web/JavaScript/Reference/Global_Objects/Promise).
```

Reviewer: Pensei em manter "Promise", porque a página sobre promises em português manteve em inglês e também porque geralmente nós falamos promise quando estamos desenvolvendo.

Reviewer: Os links sempre devem apontar para a página em português. Caso ela não exista o MDN tem uma página padrão que manda o usuário para a página em inglês.

## Sintaxe

```js
let removing = browser.browsingData.removeCache(

>   removalOptions            // RemovalOptions object

Reviewer: ```suggestion
  removalOptions            // objeto RemovalOptions
```

)
```

### Parâmetros

- `removalOptions`{{optional_inline}}

>   - : `object`. A {{WebExtAPIRef("browsingData.RemovalOptions")}} object. This parameter has no effect.

Reviewer: ```suggestion
  - : `object`. Um objeto {{WebExtAPIRef("browsingData.RemovalOptions")}}. Este parâmetro não tem efeito.
```

### Valor retornado

> Uma [`Promessa`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) que será realizada sem argumentos quando a remoção for finalizada. Se ocorrer qualquer erro, a promessa será rejeitada com uma mensagem de erro.

Reviewer: ```suggestion
Uma [`Promise`](/pt-BR/docs/Web/JavaScript/Reference/Global_Objects/Promise) que será realizada sem argumentos quando a remoção for finalizada. Se ocorrer qualquer erro, a promise será rejeitada com uma mensagem de erro.
```

## Exemplos

Limpar o cache do navegador:

```js
function onRemoved() {

>   console.log("removed");

Reviewer: ```suggestion
  console.log("removido");
```

}

function onError(error) {
  console.error(error);
}

browser.browsingData.removeCache({}).
then(onRemoved, onError);
```

## Compatibilidade com navegadores

{{Compat}}

{{WebExtExamples}}

> > **Note:** This API is based on Chromium's [`chrome.browsingData`](https://developer.chrome.com/docs/extensions/reference/browsingData/) API.

Reviewer: ```suggestion
> **Nota:** Esta API é baseada na API do Chromium [`chrome.browsingData`](https://developer.chrome.com/docs/extensions/reference/browsingData/).
```

>

> > Microsoft Edge compatibility data is supplied by Microsoft Corporation and is included here under the Creative Commons Attribution 3.0 United States License.

Reviewer: ```suggestion
> A compatibilidade dos dados do Microsoft Edge é fornecida pela Microsoft Corporation e é incluída aqui sob a licença Creative Commons Attribution 3.0 United States.
```
